Source/OpalGraphics/CGPath.m

```diff
@@ -511,3 +870,127 @@ void CGPathAddEllipseInRect(
   p2 = CGPointMake(originx + width / 2 + hdiff, originy + height);
   CGPathAddCurveToPoint(path, m, p1.x, p1.y, p2.x, p2.y, p.x, p.y);
 }
+
+CGRect CGPathGetPathBoundingBox(CGPathRef path)

```

Comment:
My computer graphics lectures were a long time ago but from what I remember your algorithm seems completely wrong. What you do is compute the bounding box of the curve by leaving out some of the control points. This won't give you the actual bounds of the curve. Please have a look at http://floris.briolas.nl/floris/2009/10/bounding-box-of-cubic-bezier/ or other explanations on the web.
What you basically have to do for the arc bits is to compute enough points on the curve to determine an exact bounding box.